### PR TITLE
Pin @azure/ms-rest-js and @azure/cognitiveservices-luis-runtime packages

### LINKED ISSUE #48: Intentalyzer is incompatible with new versions of dependencies
### Issue

When trying to create a new `IntentResolver`, using `intentalyzer.createIntentResolver` and `intentalyzerLuis.createLuisIntentRecognizer`, there is an error when creating the `LUISRuntimeClient`. This error is caused by a mismatch in the interface between versions intentalyzer's dependencies and the most recent version of these dependencies.

Specifically two packages...
-`@azure/cognitiveservices-luis-runtime`
-`@azure/ms-rest-js`

The error caused by the newest version of `@azure/ms-rest-js`...
```
error TS2345: Argument of type 'import("/Users/tkerchev/projects/timer-service/node_modules/@azure/cognitiveservices-luis-runtime/esm/lUISRuntimeClient").LUISRuntimeClient' is not assignable to parameter of type 'import("/Users/tkerchev/projects/timer-service/node_modules/intentalyzer-luis-integration/node_modules/@azure/cognitiveservices-luis-runtime/esm/lUISRuntimeClient").LUISRuntimeClient'.
  Types of property 'prediction' are incompatible.
    Type 'import("/Users/tkerchev/projects/timer-service/node_modules/@azure/cognitiveservices-luis-runtime/esm/operations/predictionOperations").PredictionOperations' is not assignable to type 'import("/Users/tkerchev/projects/timer-service/node_modules/intentalyzer-luis-integration/node_modules/@azure/cognitiveservices-luis-runtime/esm/operations/predictionOperations").PredictionOperations'.
      Types have separate declarations of a private property 'client'.

53         intentalyzerLuis.createLuisIntentRecognizer(runtime, config.luisModel.appId),
```

The error caused by the newest version of `@azure/ms-rest-js`...
```
error TS2345: Argument of type 'TokenCredentials' is not assignable to parameter of type 'ServiceClientCredentials'.
  Types of property 'signRequest' are incompatible.
    Type '(webResource: import("/Users/tkerchev/projects/timer-service/node_modules/@azure/ms-rest-js/es/lib/webResource").WebResource) => Promise<import("/Users/tkerchev/projects/timer-service/node_modules/@azure/ms-rest-js/es/lib/webResource").WebResource>' is not assignable to type '(webResource: import("/Users/tkerchev/projects/timer-service/node_modules/intentalyzer-luis-integration/node_modules/@azure/ms-rest-js/es/lib/webResource").WebResource) => Promise<...>'.
      Types of parameters 'webResource' and 'webResource' are incompatible.
        Type 'import("/Users/tkerchev/projects/timer-service/node_modules/intentalyzer-luis-integration/node_modules/@azure/ms-rest-js/es/lib/webResource").WebResource' is not assignable to type 'import("/Users/tkerchev/projects/timer-service/node_modules/@azure/ms-rest-js/es/lib/webResource").WebResource'.
          Types of property 'headers' are incompatible.
            Type 'import("/Users/tkerchev/projects/timer-service/node_modules/intentalyzer-luis-integration/node_modules/@azure/ms-rest-js/es/lib/httpHeaders").HttpHeaders' is not assignable to type 'import("/Users/tkerchev/projects/timer-service/node_modules/@azure/ms-rest-js/es/lib/httpHeaders").HttpHeaders'.
              Types have separate declarations of a private property '_headersMap'.

45         new msRest.TokenCredentials(config.luisModel.authToken),
```

The code to cause both of these errors is as follows (imports omitted)...

```
/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
export function createIntentResolver(config: IntentResolverConfiguration): IIntentResolver<any, ResolvedEntity> {
    const runtime = new luisRuntime.LUISRuntimeClient(
        new msRest.TokenCredentials(config.luisModel.authToken),
        config.luisModel.endpointUrl,
    );

    return intentalyzer.createIntentResolver(
        intentalyzerLuis.createLuisIntentRecognizer(runtime as luisRuntime.LUISRuntimeClient, config.luisModel.appId),
        intentalyzer.createTransformPipeline([sortEntitiesByUtteranceOffsetStartIndexTransform]),
    );
}
```

### Workaround

To get around these errors, I pin the version of these two packages (in the service I am writing) to the ones used by `intentalyzer-luis-integration`.

```
@azure/cognitiveservices-luis-runtime: 3.0.1
@azure/ms-rest-js: 1.8.13
```

Once these versions align, it works!

### PR DESCRIPTION
**Pull request by SpaceKatt**
Fixes #48 , see the issue for a detailed description of bug. 

tldr; `@azure/ms-rest-js` was causing issues because of differing versions between package dependencies.

Also fixes some type discrepancies, i.e. the following error...
```ts
$ npm run build

> intentalyzer-luis-integration@0.4.1-beta build /Users/tkerchev/spacekatt_projects/domain-specific-entities-node/packages/luis-integration
> tsc --build .

src/types.ts:48:18 - error TS2320: Interface 'LuisCompositeEntity' cannot simultaneously extend types 'LuisEntityBase' and 'CompositeEntity<LuisBasicEntity>'.
  Named property 'utteranceOffsets' of types 'LuisEntityBase' and 'CompositeEntity<LuisBasicEntity>' are not identical.

48 export interface LuisCompositeEntity extends LuisEntityBase, CompositeEntity<LuisBasicEntity> {
                    ~~~~~~~~~~~~~~~~~~~


Found 1 error.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! intentalyzer-luis-integration@0.4.1-beta build: `tsc --build .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the intentalyzer-luis-integration@0.4.1-beta build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/tkerchev/.npm/_logs/2020-02-03T18_49_51_075Z-debug.log
```

### CODE REVIEW

File: packages/intentalyzer/src/entities.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export interface Entity {
     readonly utteranceOffsets?: {
         startIndex: number;
         endIndex: number;
+        length?: number;
```
Comment (drub0y) [inline packages/intentalyzer/src/entities.ts:7]: Trying to figure out how this got in here...? I remember doing some work right after M2 to add this and I honestly forgot where it left off, but I'm wondering why it's showing up here in this PR now?

File: packages/luis-integration/src/types.ts [hunk 2]
```diff
@@ -17,10 +17,10 @@ export interface LuisPredictionCallContext {
 interface LuisEntityBase extends Entity {
     readonly $raw: any;
     readonly value: any;
-    readonly utteranceOffsets: {
+    readonly utteranceOffsets?: {
         startIndex: number;
         endIndex: number;
-        length: number;
+        length?: number;
```
Comment (drub0y) [inline packages/luis-integration/src/types.ts:23]: Related/same as above:

Trying to figure out how this got in here...? I remember doing some work right after M2 to add this and I honestly forgot where it left off, but I'm wondering why it's showing up here in this PR now?

**Review by drub0y — CHANGES_REQUESTED** [review]

File: packages/intentalyzer/src/entities.ts (hunk 1 above)
Comment (SpaceKatt) [reply packages/intentalyzer/src/entities.ts:7]: Honestly, I don't remember. I just needed to add this as an optional field for the project to compile

P.S.: sorry for the delayed response, I've been moving this week so my life is in boxes

Comment (SpaceKatt) [reply packages/intentalyzer/src/entities.ts:7]: It seems redundant, since it can be calculated from the difference of `startIndex` and `endIndex`. The compile time error is shown in the PR description
